### ISSUE TITLE
[Paper Note] Skill-Targeted Adaptive Training, Yinghui He+, arXiv'25, 2025.10

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#3346]
# URL
- https://arxiv.org/abs/2510.10023
# Authors
  - Yinghui He
  - Abhishek Panigrahi
  - Yong Lin
  - Sanjeev Arora
# Abstract
  - Language models often show little to no improvement (i.e., "saturation") when trained via vanilla supervised fine-tuning (SFT) on data similar to what they saw in their training set (e.g., MATH). We introduce a new fine-tuning strategy, STAT, to train such a student model by using the metacognition ability of a stronger large language model (LLM) as the teacher. The teacher uses the task dataset to create a list of skills needed for the task, and then labels each data point with its required skills (Didolkar et al., 2024). By monitoring the student's answers, the teacher creates a Missing-Skill-Profile for the student, tracking how often they failed to apply each skill in their responses. We use this idea to build a modified training set in one of two ways. In STAT-Sel, the teacher uses an existing set of training examples but adaptively reweights them according to the Missing-Skill-Profile. In STAT-Syn, the teacher synthesizes additional examples involving missing skills. Across extensive experiments on Llama and Qwen models, our methods yield improvements of up to 7.5% on MATH, whereas SFT provides only limited gains. Furthermore, STAT enhances performance on out-of-distribution benchmarks (e.g., AIME24/25, AMC23, etc.) by an average of 4.6%. Crucially, we find that STAT is complementary to RL via GRPO (Shao et al., 2024): after the model is improved using STAT to address skill gaps, GRPO continues to add further gains. We conclude that skill-targeted adaptive training should broadly improve current training pipelines. Our code is available at: https://github.com/princeton-pli/STAT.
# Translation (by gpt-4o-mini)
- 言語モデルは、トレーニングセットで見たデータに類似したデータ（例：MATH）でバニラの教師ありファインチューニング（SFT）を行った場合、ほとんど改善が見られない（すなわち「飽和」）ことがよくあります。本研究では、より強力な大規模言語モデル（LLM）のメタ認知能力を教師として利用し、そのような学生モデルを訓練するための新しいファインチューニング戦略、STATを提案します。教師はタスクデータセットを使用して、タスクに必要なスキルのリストを作成し、各データポイントに必要なスキルをラベル付けします（Didolkar et al., 2024）。学生の回答を監視することで、教師は学生のMissing-Skill-Profileを作成し、学生が各スキルを応用できなかった頻度を追跡します。このアイデアを用いて、2つの方法のいずれかで修正されたトレーニングセットを構築します。STAT-Selでは、教師は既存のトレーニング例のセットを使用しますが、Missing-Skill-Profileに基づいてそれらの重みを適応的に再調整します。STAT-Synでは、教師が欠落しているスキルを含む追加の例を合成します。LlamaおよびQwenモデルに関する広範な実験を通じて、我々の手法はMATHで最大7.5%の改善をもたらし、一方でSFTは限られた向上しか提供しません。また、STATは分布外ベンチマーク（例：AIME24/25、AMC23など）で平均4.6%のパフォーマンス向上を実現します。重要なことに、STATはGRPOを介した強化学習（Shao et al., 2024）と補完的であることがわかりました：モデルがスキルのギャップを解消するためにSTATを使用して改善された後、GRPOはさらなる向上を加え続けます。私たちは、スキルターゲットの適応トレーニングが現在のトレーニングパイプラインを広く改善するべきであると結論付けます。私たちのコードは以下のリンクで入手可能です：
https://github.com/princeton-pli/STAT。
# Summary (by gpt-4o-mini)
- 本研究では、言語モデルのメタ認知能力を活用した新しいファインチューニング戦略「STAT」を提案。教師モデルがタスクに必要なスキルをラベル付けし、学生モデルのスキル不足を追跡することで、トレーニングセットを修正。STAT-Selでは既存の例の重みを調整し、STAT-Synでは新たな例を合成。実験により、MATHで最大7.5%の改善を達成し、分布外ベンチマークでも平均4.6%の向上を示した。STATは強化学習手法GRPOと補完的であり、スキルターゲットの適応トレーニングがトレーニングパイプラインを改善することを示唆。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/yinghui_he_/status/1980257694704619679?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
関連:
- #3361

**Comment by AkihikoWatanabe** [comment]
Reward Modelでquestionがeasy/hardを定量化し、hardなものに対してモデルが応答を生成。応答の結果をstronger modelに確認させ、モデルにどのようなスキルが不足しているかを特定する。これによりモデルのスキルに関するprofileが作成されるのでこれに基づいて学習データの各サンプルとスキルを紐づけた上でサンプルを重みの調整、および不足しているスキルに関するデータを合成しSFTに活用する、といった話な模様。

<img width="838" height="496" alt="Image" src="https://github.com/user-attachments/assets/8f5e9efb-c096-4897-8327-daed9e4c920a" />

結果を見ると、+SFT / +GRPOよりも性能が高くなっている。Table1ではLlamaでの結果しか掲載されていないが、Qwenでも実験がされて同様の結果が得られている。
<img width="856" height="255" alt="Image" src="https://github.com/user-attachments/assets/d35077ef-bf33-4c12-82e1-37cbc40247af" />

また、Figure4を見ると不足していたスキルが学習によってきちんと補われていることが分かる。

<img width="839" height="541" alt="Image" src="https://github.com/user-attachments/assets/bd07a4e5-87c8-4ab1-a45c-379cff343e33" />

（評価と考察部分をもう少しじっくり読みたい）
